src/test/java/edu/berkeley/ground/db/CassandraResultsTest.java

```diff
@@ -0,0 +1,184 @@
+package edu.berkeley.ground.db;
+
+import com.datastax.driver.core.DataType;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.exceptions.CodecNotFoundException;
+import com.google.common.reflect.TypeToken;
+import edu.berkeley.ground.exceptions.GroundDbException;
+import org.junit.Test;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import static org.mockito.Mockito.*;
+
+public class CassandraResultsTest {
+  @Test
+  public void shouldGetStringFromValidResultSet() throws GroundDbException {
+    ResultSet rs = mock(ResultSet.class);
+    Row row = mock(Row.class);
+    when(row.getString("field")).thenReturn("value");
+    when(rs.one()).thenReturn(row);
+    CassandraResults results = new CassandraResults(rs);
+    assertThat(results.getString("field")).isEqualTo("value");
+  }
+
+  @Test(expected = GroundDbException.class)
+  public void shouldFailWhenFieldInvalid() throws GroundDbException {
+    setupInvalidField("field").getString("field");
+  }
+
+  @Test(expected = GroundDbException.class)
+  public void shouldFailWhenCodecNotFoundForString() throws GroundDbException {
+    setupCodecNotFoundForField("field").getString("field");
+  }
+
+  @Test
+  public void shouldGetBooleanFromValidResultSet() throws GroundDbException {
+    ResultSet rs = mock(ResultSet.class);
+    Row row = mock(Row.class);
+    when(row.getString("trueField")).thenReturn("true");
+    when(row.getString("falseField")).thenReturn("false");
+    when(row.getString("trueCapitalizedField")).thenReturn("True");
+    when(row.getString("falseCapitalField")).thenReturn("FALSE");
+    when(rs.one()).thenReturn(row);
+    CassandraResults results = new CassandraResults(rs);
+    assertThat(results.getBoolean("trueField")).isEqualTo(true);
+    assertThat(results.getBoolean("falseField")).isEqualTo(false);
+    assertThat(results.getBoolean("trueCapitalizedField")).isEqualTo(true);
+    assertThat(results.getBoolean("falseCapitalField")).isEqualTo(false);
+  }
+
+  @Test(expected = GroundDbException.class)
+  public void shouldFailWhenCodecNotFoundForBoolean() throws GroundDbException {
+    setupCodecNotFoundForField("field").getBoolean("field");
+  }
+
+  @Test(expected = GroundDbException.class)
+  public void shouldFailWhenFieldInvalidForBoolean() throws GroundDbException {
+    setupInvalidField("field").getBoolean("field");
+  }
+
+  // Should this be expected to fail? - it does not!
```

Comment:
you're right. It's expected behavior. I've removed the commented out test and added an assert to `shouldGetBooleanFromValidResultSet()`